src/v/cluster/security_frontend.cc

```diff
@@ -322,24 +325,35 @@ security_frontend::get_bootstrap_user_creds_from_env() {
         return {};
     }
 
-    ss::sstring creds_str = creds_str_ptr;
-    auto colon = creds_str.find(":");
-    if (colon == ss::sstring::npos || colon == creds_str.size() - 1) {
+    std::string_view creds = creds_str_ptr;
+    std::vector<std::string_view> parts;
+    parts.reserve(3);
+    boost::algorithm::split(parts, creds, [](char c) { return c == ':'; });
+    if (
+      parts.size() != 2 || parts.size() != 3 || parts[0].empty()
+      || parts[1].empty()) {
         // Malformed value.  Do not log the value, it may be malformed
         // but it is still a secret.
         vlog(
           clusterlog.warn,
-          "Invalid value of {} (expected \"username:password\")",
+          "Invalid value of {} (expected \"username:password:mechanism\")",
           bootstrap_user_env_key);
         return {};
     }
-
-    auto username = security::credential_user{creds_str.substr(0, colon)};
-    auto password = creds_str.substr(colon + 1);
-    auto credentials = security::scram_sha256::make_credentials(
-      password, security::scram_sha256::min_iterations);
+    std::variant<security::scram_sha256, security::scram_sha512> scram;
+    if (
+      parts.size() == 3
```

Comment:
It looks like this second conditional `if (parts.size() == 3 ...` can never be reached, because if `parts.size()` is `3`, then `parts.size() != 2` in the first conditional will be true and the warning will be logged and control will return.

it kind of looks like the first conditional should just be `if parts.size() != 3` ?

Am I missing something?

Reply:
You're right there is an issue, but this should allow for the size being either 2 or 3. I've pushed an update which should resolve this: https://github.com/redpanda-data/redpanda/pull/13550/files#diff-eff84f6ac14c3c51ab8790b3386ca78baf7bef9a22ad595aa1ec34dd4d959dacR333

Reply:
Ooops, that was my fault!